internal/appsec/dyngo/instrumentation/httpsec/tags.go

```diff
@@ -107,43 +106,56 @@ func ippref(s string) *instrumentation.NetaddrIPPrefix {
 	return nil
 }
 
-// ClientIPTags generates the IP related span tags for a given request headers
-func ClientIPTags(hdrs map[string][]string, remoteAddr string) (tags map[string]string, clientIP instrumentation.NetaddrIP) {
-	tags = map[string]string{}
+// ClientIPTags returns the resulting Datadog span tags `http.client_ip`
+// containing the client IP and `network.client.ip` containing the remote IP.
+// The tags are present only if a valid ip address has been returned by
+// ClientIP().
+func ClientIPTags(hdrs map[string][]string, hasCanonicalMIMEHeaderKeys bool, remoteAddr string) (tags map[string]string, clientIP instrumentation.NetaddrIP) {
+	remoteIP, clientIP := ClientIP(hdrs, hasCanonicalMIMEHeaderKeys, remoteAddr)
+
+	remoteIPValid := remoteIP.IsValid()
+	clientIPValid := clientIP.IsValid()
+	if !remoteIPValid && !clientIPValid {
+		return nil, instrumentation.NetaddrIP{}
+	}
+
+	tags = make(map[string]string, 2)
+	if remoteIPValid {
+		tags["network.client.ip"] = remoteIP.String()
+	}
+	if clientIPValid {
+		tags[ext.HTTPClientIP] = clientIP.String()
+	}
+
+	return tags, clientIP
+}
+
+// ClientIP returns the first public IP address found in the given headers. If
+// none is present, it returns the first valid IP address present, possibly
+// being a local IP address. The remote address, when valid, is used as fallback
+// when no IP address has been found at all.
+func ClientIP(hdrs map[string][]string, hasCanonicalMIMEHeaderKeys bool, remoteAddr string) (remoteIP, clientIP instrumentation.NetaddrIP) {
```

Comment:
Minor nit: maybe we can rename `hasCanonicalMIMEHeaderKeys` to something a bit shorter?
Something like `hasCanonicalHeaders` or even just `canonicalHeaders`?
No strong opinion there so I'm also fine leaving it as is.